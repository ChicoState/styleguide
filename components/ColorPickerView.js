import React, { useState } from 'react'
import { View, Text } from 'react-native'
import { TriangleColorPicker, toHsv } from 'react-native-color-picker'
import { SG_Color, SGCFromHSV, HSVTORGB } from '../utils/colorHelpers'

export default function ColorPickerView({ navigation }) {
  const [SelColor, SetSelColor] = useState(toHsv('green')) // initalize with green

  // onColorChange = (color) => {
  //   SetSelColor(color);
  // }

  return (
    <View style={{ flex: 1, padding: 45, backgroundColor: '#212021' }}>
      <Text style={{ color: 'white' }}>React Native Color Picker - Controlled</Text>
      <TriangleColorPicker
        oldColor='purple'
        color={SelColor}
        onColorChange={(color) => SetSelColor(color)}
        // HERE
        onColorSelected={color => navigation.navigate('Recommendations', {
          SelectedColor: toHsv(color)
        })}
        // onColorSelected={color => alert(`Color selected: ${color}`)}
        // onOldColorSelected={color => alert(`Old color selected: ${color}`)}
        style={{ flex: 1 }}
      />
    </View>
  )
}


