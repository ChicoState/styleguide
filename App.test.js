import React from 'react';
import renderer from 'react-test-renderer';
import CameraView from './components/cameraView'
import RecommendationView from './components/recommendationView'
import MainView from './components/mainView'
import ColorPickerView from './components/ColorPickerView';

import { render, screen, fireEvent } from '@testing-library/react-native';
import { StyleSheet, Text, View, Button, Image, SafeAreaView, TouchableWithoutFeedback } from 'react-native';
describe('CameraView', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<CameraView />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});
describe('MainView', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<MainView />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});
describe('ColorPickerView', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<ColorPickerView />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});
// Path: App.test.js
