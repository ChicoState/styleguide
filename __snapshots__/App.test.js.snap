// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPickerView renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#121212",
      "flex": 1,
      "padding": 45,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    React Native Color Picker - Controlled
  </Text>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "height": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "flex": 1,
            "opacity": 1,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#008000",
            "flex": 1,
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`MainView renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "alignItems": "center",
        "color": "black",
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": "40%",
        "marginTop": "50%",
      }
    }
  >
    Style Guide
  </Text>
</View>
`;
